runtime/doc/channels.txt

```diff
@@ -0,0 +1,91 @@
+*channel.txt*    Nvim
+
+
+		 NVIM REFERENCE MANUAL    by Thiago de Arruda
+
+
+Nvim's facilities for async io				  *channels*
+
+				      Type <M-]> to see the table of contents.
+
+==============================================================================
+1. Introduction						    *channels-intro*
+
+Channels are nvim's way of communicating with external processes.
+
+There are several ways to open a channel:
+
+  1. Through stdin/stdout when `nvim` is started with `--headless`, and a startup
+     script or --cmd  commad opens the stdio channel using |stdioopen()|.
+
+  2. Through stdin, stdout and stderr of a process spawned by |jobstart()|.
+
+  3. Through the PTY master end of a PTY opened with
+     `jobstart(..., {'pty': v:true})` or |termopen()|.
+
+  4. By connecting to a TCP/IP socket or named pipe with |sockconnect()|.
+
+  5. By another process connecting to a socket listened by nvim. This only
+     supports RPC channels, see |rpc-connecting|.
+
+Channels can operate in two different modes, a mode using the |rpc| protocol,
```

Comment:
I would re-organize to:

Channels can operate in two modes: one via the |rpc| protocol (msgpack-rpc
based) and the other in "bytes" mode where scripts read and write raw bytes
over the channel. As a caveat even if a job's channel is in |rpc| mode, bytes can still
be read over its' stderr.  Additionally only bytes can be written to nvim's stderr.

(Honestly it doesn't make a big difference either way, feel completely free to reject
this suggestion).